app/src/androidTest/java/com/github/se/travelpouch/ui/dashboard/AddActivityScreenTest.kt

```diff
@@ -0,0 +1,160 @@
+        .addActivity(anyOrNull(), anyOrNull(), anyOrNull())
+
+    // verify no saving when blank description
+    completeAllFields(composeTestRule)
+    composeTestRule.onNodeWithTag("descriptionField").performTextClearance()
+    composeTestRule.onNodeWithTag("saveButton").performClick()
+    verify(mockActivityRepositoryFirebase, never())
+        .addActivity(anyOrNull(), anyOrNull(), anyOrNull())
+
+    // verify no saving when blank date
+    completeAllFields(composeTestRule)
+    composeTestRule.onNodeWithTag("dateField").performTextClearance()
+    composeTestRule.onNodeWithTag("saveButton").performClick()
+    verify(mockActivityRepositoryFirebase, never())
+        .addActivity(anyOrNull(), anyOrNull(), anyOrNull())
+  }
+
+  @Test
+  fun doesSaveWhenFieldsAreFull() {
+    composeTestRule.setContent { AddActivityScreen(mockActivityModelView) }
+    completeAllFields(composeTestRule)
+    composeTestRule.onNodeWithTag("saveButton").performClick()
+    verify(mockActivityRepositoryFirebase).addActivity(anyOrNull(), anyOrNull(), anyOrNull())
+  }
+
+  @Test
+  fun dateFormattingWorksCorrectly() {
+    composeTestRule.setContent { AddActivityScreen(mockActivityModelView) }
+    composeTestRule.onNodeWithTag("dateField").performTextClearance()
+    composeTestRule.onNodeWithTag("dateField").performTextInput("00000000")
+    val result =
+        composeTestRule.onNodeWithTag("dateField").fetchSemanticsNode().config[EditableText]
+    assert(result.text == "00/00/0000")
+  }
+
+  @Test
+  fun doesNotSaveWhenDateIsWrong() {
+    composeTestRule.setContent { AddActivityScreen(mockActivityModelView) }
+    completeAllFields(composeTestRule)
+    composeTestRule.onNodeWithTag("dateField").performTextClearance()
+    composeTestRule.onNodeWithTag("dateField").performTextInput("00000000")
+
+    composeTestRule.onNodeWithTag("saveButton").performClick()
+    verify(mockActivityRepositoryFirebase, never())
+        .addActivity(anyOrNull(), anyOrNull(), anyOrNull())
+  }
+
+  @Test
+  fun noCharacterAllowedInDateField() {
+    composeTestRule.setContent { AddActivityScreen(mockActivityModelView) }
+    composeTestRule.onNodeWithTag("dateField").performTextClearance()
+    composeTestRule.onNodeWithTag("dateField").performTextInput("mdkdk")
+    var result =
+        composeTestRule.onNodeWithTag("dateField").fetchSemanticsNode().config[EditableText]
+    assert(result.text == "")
+    composeTestRule.onNodeWithTag("dateField").performTextInput("l09kfj89iuiui2025")
+    result = composeTestRule.onNodeWithTag("dateField").fetchSemanticsNode().config[EditableText]
+    assert(result.text == "")
+    composeTestRule.onNodeWithTag("dateField").performTextClearance()
+    composeTestRule.onNodeWithTag("dateField").performTextInput("1234")
+    composeTestRule.onNodeWithTag("dateField").performTextInput("l")
+    result = composeTestRule.onNodeWithTag("dateField").fetchSemanticsNode().config[EditableText]
+    assert(result.text == "12/34/")
+  }
```

Comment:
In the tests, you're checking for invalid input cases, which is great, but it would also be good to test when everything is correct and ensure that it properly calls addActivity

Reply:
Actually, there a test, that is a bit hidden but it checks everything works when the fields are valid. It is called "doesSaveWhenFieldsAreFull" at lines 103 to 107.

Reply:
Yes you are right